scripts/release-docs.ts

```diff
@@ -0,0 +1,34 @@
+import { execSync } from 'child_process';
+
+// The GITHUB_REF_NAME is a full version (i.e. 17.3.2). The branchName will strip the patch version number.
+// We will publish docs to the website branch based on the current tag (i.e. website-17)
+console.log(`Comparing ${process.env.GITHUB_REF_NAME} to npm versions`);
+
+const majorVersion = process.env.GITHUB_REF_NAME?.split('.')[0];
+const minorVersion = process.env.GITHUB_REF_NAME?.split('.')
+  .slice(0, 2)
+  .join('.');
+const releasedVersions: string[] = JSON.parse(
+  execSync(`npm show nx@^${majorVersion} version --json`).toString()
+);
+const versionIsReleased =
+  minorVersion &&
+  releasedVersions.some((version) => version.includes(minorVersion));
+const latestVersion = releasedVersions.slice().sort().pop();
+const versionIsLatest = minorVersion && latestVersion?.includes(minorVersion);
```

Comment:
I think there's a gap in the logic here.

If the version being published is... `21.1.1`.. `minorVersion` here is `1.1` right? And if the `latestVersion` was `21.11.1`, it would include `1.1`. Both at the end.. and in the middle. 

This would misidentify the `21.1.1` publish as being latest and unintentionally publish the docs as `latest` right?

Now that this is a Node script we can leverage `semver` to do this check. I think it would be easier to read and be more accurate.

Idk if the above scenario will happen anytime soon but could you take another look at this logic so it's more bullet proof?

Reply:
`minorVersion` would be `21.1`, but yes the string comparison isn't bullet proof.  I'll use the `semver` library